api/src/controllers/put/administrative-penalty.js

```diff
@@ -110,19 +111,48 @@ exports.editRecord = async function(args, res, next, incomingObj) {
 
   try {
     savedAdministrativePenalty = await this.editMaster(args, res, next, incomingObj, flavourIds);
-
+  } catch (e) {
     return {
-      status: 'success',
+      status: 'failure',
       object: savedAdministrativePenalty,
-      flavours: savedFlavourAdministrativePenalties
+      errorMessage: e.message
     };
+  }
+
+  // update document roles
+  let documentPromises = [];
+  let savedDocuments = [];
+
+  try {
+    const isAnonymous = QueryUtils.isRecordAnonymous(savedAdministrativePenalty);
+
+    if (isAnonymous) {
+      savedAdministrativePenalty.documents.forEach(docId => {
+        documentPromises.push(DocumentController.unpublishDocument(docId, args.swagger.params.auth_payload));
+      });
+    } else if (savedAdministrativePenalty) {
+      savedAdministrativePenalty.documents.forEach(docId => {
+        DocumentController.publishDocument(savedAdministrativePenalty._id, docId, args.swagger.params.auth_payload);
+      });
+    }
```

Comment:
Since you're not using isAnonymous anywhere else, I'd rather save some vertical space and use:
`if (QueryUtils.isRecordAnonymous(savedAdministrativePenalty)) {`

There's a few of these in the PR.

Reply:
This looks a bit weird, possibly I did a find/replace when I made some updates, and it was left looking a bit odd.  I can update this.